### ISSUE TITLE
Softmax, LogSoftmax are over parameterized

### ISSUE THREAD

**Issue by chiragnagpal** [pytorch/pytorch#78450]
### 🚀 The feature, motivation and pitch

The current implementation of Softmax and LogSoftmax for a K class output involves an [NxK] input tensor. This is over parameterized and leads to issues of identifiability. It would be worthwhile to include an implementation that only requires an input tensor [NX(K-1)] dimensionality.

**Comment by jbschlosser** [comment]
Hey @chiragnagpal, thanks for the request! Is the idea to accept an input of shape `(N, K-1)` and treat the final element as an implicit zero to produce a softmax output of shape `(N, K)`?
